Src/IronPython/Runtime/Operations/StringOps.cs

```diff
@@ -1226,7 +1226,7 @@ public static int rindex([NotNull]this string self, [NotNull]string sub, object?
                 var idx = self[i];
                 try {
                     ret.AppendValueForTranslate(PythonCalls.Call(context, getitem, (int)idx));
-                } catch (Exception e) when (e is LookupException || e is IndexOutOfRangeException || e is KeyNotFoundException) {
+                } catch (Exception e) when (e is LookupException || e is IndexOutOfRangeException || e is KeyNotFoundException || PythonOps.IsInstance(e.GetPythonException(), PythonExceptions.LookupError)) {
```

Comment:
It appears that `IsInstance` covers the other three tests as well, if you are looking for simplifying the condition.

Reply:
Unfortunately it doesn't cover the other three cases if the exception is thrown from C#. The underlying Python exception isn't created until it's needed (so `e.GetPythonException()` can be `null`). It looks like I could use `PythonExceptions.ToPython(e)` which would instantiate the Python exception, but that would make an already super slow code path even slower.

Hopefully code that takes this overload won't have too many misses. In my case it came up in the stdlib where it uses a list that covers every character so I guess it won't be an issue.

Reply:
I see. I've checked the code parts known to me and I do not see any other places where such catch would be necessary, so perhaps this is an outlier. (Otherwise I was thinking about a static test method somewhere, maybe in `LookupException`).

Reply:
Here's one:
```Python
class Test(SystemExit): pass

raise Test
```

Anyway, will merge this as-is for now and file a separate issue.